src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php

```diff
@@ -177,25 +182,44 @@ private function isOtherServerProcessRunning($address)
     }
 
     /**
+     * Determine the absolute file path for the router script based on user input and the environment.
+     *
+     * @param string|null     $router File path of the custom router script, if set by the user; otherwise null
+     * @param string          $env    The application environment
+     * @param OutputInterface $output An OutputInterface instance
+     *
+     * @return string|bool The absolute file path of the router script, or false on failure
+     */
+    private function determineRouterScript($router, $env, OutputInterface $output)
+    {
+        if (null === $router) {
+            $router = $this
+                ->getContainer()
+                ->get('kernel')
+                ->locateResource(sprintf('@FrameworkBundle/Resources/config/router_%s.php', $env))
+            ;
+        } elseif (!file_exists($router)) {
+            $output->writeln(sprintf('<error>The given router script "%s" does not exist</error>', $router));
+
+            return false;
+        }
+
+        return realpath($router);
```

Comment:
@abulford What do you think if we changed it a bit like this:

```php
if (null === $router) {
    $router = $this
        ->getContainer()
        ->get('kernel')
        ->locateResource(sprintf('@FrameworkBundle/Resources/config/router_%s.php', $env));
}

if (false === $path = realpath($router)) {
    $output->writeln(sprintf('<error>The given router script "%s" does not exist</error>', $router));

    return false;
}

return $path;
```

We then won't have to call `file_exists()` at all and we would also always show an error message if the file doesn't exist.